operators/azure/logger.hpp

```diff
@@ -0,0 +1,95 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#pragma once
+
+#include "ocos.h"
+
+// may or may not be available depending on the ORT version
+struct OrtLogger;  // may or may not exist
+
+namespace ort_extensions {
+// Disable GCC 'ignoring attributes on template argument' warning due to Logger_LogMessage using
+// `__attribute__((warn_unused_result))`.
+// The template here is about whether Logger_LogMessage exists or not, so the attribute is irrelevant.
+#ifdef __GNUC__
+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Wignored-attributes"
+#endif
+
+template <class, class = void>
+struct has_Logger_LogMessage : std::false_type {};
```

Comment:
maybe move `has_Logger_LogMessage` and `LoggerImpl` into an internal `detail` namespace?